Small improvement in maintenance scheduler

MaxMorais opened the pull request:
This improvement is intended to start disable blocking Maintenance Schedule submit, when the `Sales Person` does not have a `user_id` in the Employee!

The old logic try to use the `sales_person` email_id or the `owner` to create the event, but due some fails in the code, it dont occurs!

MaxMorais committed:
Small improvement in maintenance scheduler

This improvement is intended to start disable blocking Maintenance Schedule submit, when the `Sales Person` does not have a `user_id` in the Employee!

The old logic try to use the `sales_person` email_id or the `owner` to create the event, but due some fails in the code, it dont occurs!

rmehta commented on erpnext/support/doctype/maintenance_schedule/maintenance_schedule.py:
```diff
@@ -49,26 +49,28 @@ def on_submit(self):
+				try:
+					email_map[d.sales_person] = sp.get_email_id()
+				except frappe.ValidationError:
+					pass
```
Maybe it should send a msgprint saying the email ID is not valid and then pass. If it fails silently then the user might expect the event to be made.

MaxMorais replied:
@rmehta you are right I was thiking the same!

MaxMorais replied:
@rmehta Done!

MaxMorais committed:
Added warning

Added a warning to show a information to the active user

MaxMorais committed:
Reduce start and end date verification

Given the actual scenario, that verify if the start_date <= end_date, do schedule a Maintenance, 2 days are required, but in many cases only one day is required!

MaxMorais committed:
Translation improvements

Translation improvements

rmehta commented on erpnext/support/doctype/maintenance_schedule/maintenance_schedule.py:
```diff
@@ -47,28 +47,37 @@ def on_submit(self):
+					
+			if no_email_sp:
+				frappe.msgprint(
+					frappe._("Setting Events to <b>{0}</b>, since the Employee attached to the below Sales Persons does not have a User ID<br>{1}").format(
```
Cool, lets leave out the HTML `<b>` `<br>` tags in the message. Not good for translations.

MaxMorais replied:
:+1:

MaxMorais committed:
Removed HTML tags from messages

Remove HTML from messages
